eng/packages/http-client-csharp-mgmt/generator/Azure.Generator.Management/src/Providers/ResourceClientProvider.cs

```diff
@@ -350,9 +350,24 @@ protected override MethodProvider[] BuildMethods()
                 }
                 else
                 {
-                    operationMethods.Add(new ResourceOperationMethodProvider(this, ContextualPath, restClientInfo, method, convenienceMethod, false));
+                    // For Create operations, use "CreateOrUpdate" method name instead of spec operation name
+                    string? methodName = null;
+                    if (methodKind == ResourceOperationKind.Create)
+                    {
+                        methodName = "CreateOrUpdate";  // For sync method
+                    }
+
+                    operationMethods.Add(new ResourceOperationMethodProvider(this, ContextualPath, restClientInfo, method, convenienceMethod, false, methodName));
                     var asyncConvenienceMethod = restClientInfo.RestClientProvider.GetConvenienceMethodByOperation(method.Operation, true);
-                    operationMethods.Add(new ResourceOperationMethodProvider(this, ContextualPath, restClientInfo, method, asyncConvenienceMethod, true));
+
+                    // For async method, use "CreateOrUpdateAsync"
+                    string? asyncMethodName = null;
+                    if (methodKind == ResourceOperationKind.Create)
+                    {
+                        asyncMethodName = "CreateOrUpdateAsync";  // For async method
+                    }
```

Comment:
this looks like a little bit duplicated.
could we introduce a method for this, taking an `isAsync` parameter to give us the method name? And if we do this, we only have one hard-coded name `CreateOrUpdate`